src/rgw/motr/gc/gc.h

```diff
@@ -70,6 +127,8 @@ class MotrGC : public DoutPrefixProvider {
 
   void start_processor();
   void stop_processor();
+  void enqueue(const DoutPrefixProvider* dpp, std::string iname, struct motr_gc_obj_info obj);
+  void dequeue(const DoutPrefixProvider* dpp, std::string iname, struct motr_gc_obj_info obj);
```

Comment:
Hi, do we need these methods?? 
in `entry()` we will go through each index. For every entry in the index we will call some sal function like `delete_obj` or `MotrObject::remove_mobj_and_index_entry()` and then delete the entry on success.
I am referring `RGWGC::GCWorker::entry()` and `process()` methods, can we keep the structure like them, or do you have some idea around these methods?

Reply:
Yes, the worker thread processing structure will remain the same. These functions are for metadata.
When delete is called, the s3 delete function will delete the object metadata and call the `enqueue` function to put object metadata in GC index.
When an object is successfully deleted by GC thread, it will call `dequeue` function to delete metadata entries of that object.

Reply:
Okay makes sense, but as per rados gc & lc, they have kept the enqueue method with the producer.
The consumer i.e. GC will only remove the entries from the list.

Reply:
I have done a very basic implementation of dequque as if now and removed enqueue from here.